packages/nextjs/src/config/loaders/dataFetchersLoader.ts

```diff
@@ -0,0 +1,137 @@
+import { logger } from '@sentry/utils';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { isESM } from '../../utils/isESM';
+import {
+  AST,
+  findAvailibleAlias,
+  findDeclarations,
+  findExports,
+  findIdentifiers,
+  makeAST,
+  removeComments,
+} from './ast';
+import { LoaderThis } from './types';
+
+// Map to keep track of each function's placeholder in the template and what it should be replaced with. (The latter
+// will get added as we process the user code. Setting it to an empty string here means TS won't complain when we set it
+// to a non-empty string later.)
+const DATA_FETCHING_FUNCTIONS = {
+  getServerSideProps: { placeholder: '__ORIG_GSSP__', alias: '' },
+  getStaticProps: { placeholder: '__ORIG_GSPROPS__', alias: '' },
+  getStaticPaths: { placeholder: '__ORIG_GSPATHS__', alias: '' },
+};
+
+type LoaderOptions = {
+  projectDir: string;
+};
+
+/**
+ * Find any data-fetching functions the user's code contains and rename them to prevent clashes, then whittle the
+ * template exporting wrapped versions instead down to only the functions found.
+ *
+ * @param userCode The source code of the current page file
+ * @param templateCode The source code of the full template, including all functions
+ * @param filepath The path to the current pagefile, within the project directory
+ * @returns A tuple of modified user and template code
+ */
+function wrapFunctions(userCode: string, templateCode: string, filepath: string): string[] {
+  let userAST: AST, templateAST: AST;
+  const isTS = new RegExp('\\.tsx?$').test(filepath);
```

Comment:
```suggestion
  const isTS = /\\.tsx?$/.test(filepath);
````

linters & static code analysis can work better with regex literals

Reply:
Say more about this, please. (It's not something that I've heard before.)

Reply:
When you use static code analysis tools like sonar lint. It sometimes tells you when you're doing something dumb in regex. That generally doesn't work when we're creating Regex objects via the constructor because the tools aren't smart enough to analyze the strings passed in. The regex here obviously isn't complicated enough that this matters but in other cases, it might be. Unless we're dynamically creating a regex I'd just do a literal. Also, syntax highlighting is better in literals.